infrastructure/api/sse-api.yml

```diff
@@ -317,10 +318,47 @@ Resources:
     Properties:
       Handler: backend/api/src/handlers/auth/update-client.updateClientInRegistryHandler
       Description: Updates a client using an HTTP endpoint
+      KmsKeyArn: !Ref AuthKmsKey
       Environment:
         Variables:
           AUTH_REGISTRATION_BASE_URL: !Ref AuthRegistrationBaseUrl
           AUTH_API_KEY: "{{resolve:ssm:/self-service/api/auth-api-key}}"
+      Policies:
+        - Statement:
+            - Effect: Allow
+              Action:
+                - "kms:Decrypt"
+                - "kms:Encrypt"
+              Resource:
+                - !Ref AuthKmsKey
+  
+ # KMS configuration for auth kms key
+  AuthKmsKey:
+    Type: AWS::KMS::Key
+    DeletionPolicy: Retain
+    UpdateReplacePolicy: Retain
+    # checkov:skip=CKV_AWS_7: "Ensure rotation for customer created CMKs is enabled"
+    Properties:
+      Description: Key for auth key encrypt/decrypt
+      Enabled: TRUE
+      KeyPolicy:
+        Version: '2012-10-17'
+        Statement:
+          - Sid: Enable IAM User Permissions
+            Effect: Allow
+            Principal:
+              AWS: !Sub arn:aws:iam::${AWS::AccountId}:root
+            Action: kms:*
+            Resource: '*'
+      KeySpec: ECC_NIST_P256
+      KeyUsage: ENCRYPT_DECRYPT
+
+  AuthKmsKeyAlias:
+    Type: AWS::KMS::Alias
+    Properties:
+      AliasName: !Sub "sse_auth_kms_Key"
+      TargetKeyId: !Ref AuthKmsKey
+
```

Comment:
It doesn't look like this is used anywhere?

Reply:
It is used as best practice and make it  easily to identify the key in the AWS KMS console.